The default image for the helm chart was updated to a development version of enterprise

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request. Searching for pre-existing feature requests helps us consolidate datapoints for identical requirements into a single place, thank you!
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

<!--- Thank you for keeping this note for the community --->

---

<!--- When filing a bug, please include the following headings if possible. Any example text in this template can be deleted. --->

### Overview of the Issue

We use flux and automatically update the point release of the consul helm chart in lower enviroments, e.g `1.0.x` for the consul helm chart version. Last night all of our environments with this configuration update to `1.0.8` of the helm chart. It seems like in this version the image that the chart pulls was updated from `hashicorp/consul:1.15.1` to `docker.mirror.hashicorp.services/hashicorppreview/consul-enterprise:1.16-dev` 

this causes the rollout to fail because when the new pods come up they fail because there is no enterprise license. 

It looks like this was changed in this commit 

https://github.com/hashicorp/consul-k8s/commit/37dd92944b515fd81272c31f5ddd27b88a95c5e1

### Reproduction Steps

update a non-enterprise helm install to `1.0.8`


### Logs

==> Starting Consul agent...
              Version: '1.14.9-dev+ent'
             Revision: '45189ef7'
           Build Date: '2023-06-28 23:53:06 +0000 UTC'
              Node ID: '9119d4ee-81aa-2371-cd97-d91b3956ec18'
            Node name: 'consul-server-0'
           Datacenter: 'dc1' (Segment: '<all>')
               Server: true (Bootstrap: false)
          Client Addr: [0.0.0.0] (HTTP: 8500, HTTPS: -1, gRPC: 8502, gRPC-TLS: -1, DNS: 8600)
         Cluster Addr: 10.100.3.33 (LAN: 8301, WAN: 8302)
    Gossip Encryption: true
     Auto-Encrypt-TLS: false
     Reporting Enabled: true
            HTTPS TLS: Verify Incoming: false, Verify Outgoing: false, Min Version: TLSv1_2
             gRPC TLS: Verify Incoming: false, Min Version: TLSv1_2
     Internal RPC TLS: Verify Incoming: false, Verify Outgoing: false (Verify Hostname: false), Min Version: TLSv1_2

==> Log data will now stream in as it occurs:

2023-06-30T16:32:57.033Z [WARN]  agent: bootstrap_expect > 0: expecting 5 servers
2023-06-30T16:32:57.046Z [WARN]  agent.auto_config: bootstrap_expect > 0: expecting 5 servers
2023-06-30T16:32:57.046Z [ERROR] agent: Error starting agent: error="license is missing. To add a license, configure \"license_path\" in your configuration file, use the CONSUL_LICENSE environment variable, or use the CONSUL_LICENSE_PATH environment variable. For a trial license of Consul Enterprise, visit https://
consul.io/trial."
2023-06-30T16:32:57.047Z [INFO]  agent: Exit code: code=1
Stream closed EOF for consul/consul-server-0 (consul)


### Expected behavior

<!--- What was the expected result after following the reproduction steps? --->

the helm upgrade should complete without error 

### Environment details

- `consul-k8s` version: `hashicorp/consul:1.14.7` pre update, after attempted update `docker.mirror.hashicorp.services/hashicorppreview/consul-enterprise:1.14-dev`
- `values.yaml` used to deploy the helm chart: I do not set the image in the values section 
- Kubernetes version: v1.25
- Cloud Provider (If self-hosted, the Kubernetes provider utilized): EKS

--- a/charts/consul/values.yaml
+++ b/charts/consul/values.yaml
@@ -63,7 +63,7 @@ global:
   # image: "hashicorp/consul-enterprise:1.10.0-ent"
   # ```
   # @default: hashicorp/consul:<latest version>
-  image: "docker.mirror.hashicorp.services/hashicorppreview/consul-enterprise:1.14-dev"
+  image: "hashicorp/consul:1.14.8"
 
   # Array of objects containing image pull secret names that will be applied to each service account.
   # This can be used to reference image pull secrets if using a custom consul or consul-k8s-control-plane Docker image.

